bundles/model/org.openhab.model.script/src/org/openhab/model/script/actions/ScriptExecution.java

```diff
@@ -135,4 +143,29 @@ private static Timer makeTimer(AbstractInstant instant, String closure, JobDataM
 			return null;
 		}		
 	}
+	
+	public static Timer createTimer(String cronExpression, Procedure0 closure) {
+		JobKey jobKey = new JobKey("CRON" + cronExpression + ": " + closure.toString());
+	    //Trigger trigger = newTrigger().startAt(instant.toDate()).build();
+		Trigger trigger = newTrigger()
+			.withSchedule(cronSchedule(cronExpression))
+			.build();
+			        
+			Timer timer = new TimerImpl(jobKey, trigger.getKey(), cronExpression);
+			try {
+				JobDataMap dataMap = new JobDataMap();
+				dataMap.put("procedure", closure);
+				dataMap.put("timer", timer);
+		        JobDetail job = newJob(TimerExecutionJob.class)
+		            .withIdentity(jobKey)
+		            .usingJobData(dataMap)
+		            .build();	
+		        TimerImpl.scheduler.scheduleJob(job, trigger);
+				logger.debug("Scheduled code for execution at {}", cronExpression);
+				return timer;
+			} catch(SchedulerException e) {
+				logger.error("Failed to schedule code for execution.", e);
+				return null;
+			}
+		}
```

Comment:
it looks like you've copied makeTimer() and just replaced the Trigger creation. For the sake of Don't-Repeat-Yourself (DRY) could you please extract the Trigger-Creation from the methods and use just one resulting method which assemples JobData, JobDetail and so forth